### ISSUE TITLE
Redundant Media Type */* on ControllerAdvice method with explicit ApiResponse annotation

### ISSUE THREAD

**Issue by Kiongku-Eve** [springdoc/springdoc-openapi#144]
ControllerAdvice (or RestControllerAdvice) always produces an ApiResponse with content media-type \*/\* even with an explicit ApiResponse annotation.

The following code
```
@RestControllerAdvice
public class GlobalExceptionHandler {
    @ExceptionHandler(Exception.class)
    @ResponseStatus(HttpStatus.INTERNAL_SERVER_ERROR)
    @ApiResponse(
        responseCode = "500",
        description = "Internal server error",
        content = @Content(
                mediaType = MediaType.APPLICATION_JSON_VALUE,
                schema = @Schema(implementation = ErrorDto.class)
        )
    )
    ErrorDto handleUnhandledError() {
        return ErrorDto.builder()
                .code("500")
                .description("Internal server error")
                .build();
    }
}
```
actually produces 
```
"500": {
    "description": "Internal server error",
    "content": {
        "application/json": {
            "schema": {
                "$ref": "#/components/schemas/ErrorDto"
            }
        },
        "*/*": {
            "schema": {
                "$ref": "#/components/schemas/ErrorDto"
            }
        }
    }
}
```

The desired result is
```
"500": {
    "description": "Internal server error",
    "content": {
        "application/json": {
            "schema": {
                "$ref": "#/components/schemas/ErrorDto"
            }
        }
    }
}
```

**Comment by springdoc** [comment]
fix will be available on release v1.2.4 of springdoc-openapi.

**Comment by vaibhawj** [comment]
Is it available? I am trying with v1.2.12 but the behavior is the same.
